ctrd/container.go

```diff
@@ -559,3 +564,59 @@ func (c *Client) WaitContainer(ctx context.Context, id string) (types.ContainerW
 		StatusCode: int64(msg.ExitCode()),
 	}, nil
 }
+
+// CreateCheckpoint create a checkpoint from a running container
+func (c *Client) CreateCheckpoint(ctx context.Context, id string, checkpointDir string, exit bool) error {
+	pack, err := c.watch.get(id)
+	if err != nil {
+		return err
+	}
+
+	wrapperCli, err := c.Get(ctx)
+	if err != nil {
+		return fmt.Errorf("failed to get a containerd grpc client: %v", err)
+	}
+	client := wrapperCli.client
+
+	var opts []containerd.CheckpointTaskOpts
+	if exit {
+		opts = append(opts, containerd.WithExit)
+	}
+	checkpoint, err := pack.task.Checkpoint(ctx, opts...)
+	if err != nil {
+		return fmt.Errorf("failed to checkpoint: %s", err)
+	}
+	defer client.ImageService().Delete(ctx, checkpoint.Name())
+
```

Comment:
could we split it into one function? for example, the named `getCheckpointImageDescriptor` function handled `cpDesc`.